src/components/Button/Button.tsx

```diff
@@ -0,0 +1,168 @@
+import { Button as AntButton } from 'antd'
+import classnames from 'classnames'
+
+import { ButtonHierarchies, ButtonIconPositions, ButtonShapes, ButtonSizes, ButtonTypes } from './Button.types'
+import styles from './Button.module.css'
+
+const { Primary, Neutral, Danger } = ButtonHierarchies
+const { Left, Right } = ButtonIconPositions
+const { Square } = ButtonShapes
+const { Small, Middle, Large } = ButtonSizes
+const { Filled, Ghost } = ButtonTypes
+
+export type ButtonProps = {
+
+  /**
+   * The children nodes to be rendered with the button context.
+   */
+  children?: ReactNode,
+
+  /**
+   * Defines the button hierarchy. Either:
+   *
+   * - primary: button associated with the most significant (and therefore primary) action on the page;
+   * - neutral: button associated with a general purpose action;
+   * - danger: button associated with a potentially dangerous action.
+   */
+  hierarchy?: ButtonHierarchies,
+
+  /**
+   * Redirect url of a link button.
+   */
+  href?: string,
+
+  /**
+   * Sets an icon component for the button.
+   */
+  icon?: ReactNode,
+
+  /**
+   * Defines a position for the button icon (if any). Either:
+   *
+   * - left: the icon is placed to the left of the button text;
+   * - right: the icon is placed to the right of the button text.
+   */
+  iconPosition?: ButtonIconPositions,
+
+  /**
+   * Identifies whether the button is disabled or not.
+   */
+  isDisabled?: boolean,
+
+  /**
+   * Sets the loading status of the button.
+   */
+  isLoading?: boolean,
+
+  /**
+   * Sets the handler to handle a click event.
+   */
+  onClick?: () => void,
+
+  /**
+   * Defines the button shape. Either:
+   *
+   * - square: square-shaped button with rounded corners;
+   * - circle: fully rounded button.
+   */
+  shape?: ButtonShapes,
+
+  /**
+   * Defines the button size. Either:
+   *
+   * - small: small button - 24px in height;
+   * - middle: middle button - 32px in height;
+   * - large: large button - 48px in height.
+   */
+  size?: ButtonSizes,
+
+  /**
+   * Specifies where the linked document will open when the link is clicked (default is '_blank').
+   */
+  target?: string,
+
+  /**
+   * Defines the button type. Either:
+   *
+   * - filled: button with a solid fill, typically used for primary actions;
+   * - outline: medium-emphasis button with a stroke around the button container and no fill,
+   *   typically used for secondary actions;
+   * - ghost: transparent button with no stroke, typically used for links.
+   */
+  type?: ButtonTypes,
+}
+
+/**
+ * UI component for presenting buttons
+ *
+ * @link https://ant.design/components/button
+ * @returns {Button} Button component
+ */
+export const Button = ({
+  children,
+  hierarchy,
+  href,
+  icon,
+  iconPosition,
+  isDisabled,
+  isLoading,
+  onClick,
+  shape,
+  size,
+  target,
+  type,
+}: ButtonProps): ReactElement => {
+  const buttonClassNames = useMemo(() => (
+    [
+      children && size === Small && styles['button-sm'],
```

Comment:
we felt bad here. Why not manage the class name in camelCase?

Reply:
I found online several guidelines on naming conventions for CSS classes, and all recommend using hyphens compared to camelCase. The most widespread naming convention is BEM:
https://en.bem.info/methodology/naming-convention/
https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/
https://www.makeuseof.com/css-class-and-id-best-naming-practices/

Reply:
We decided not to apply this convention because it does not fit well with the use of CSS modules